### PR TITLE
Stage 2b PR Train Step 2: analytics service + kill matrix presenter

### PR DESCRIPTION
**Pull request by davidhouweling**
## 1. context
This is Step 2 in the Stage 2b PR train for kill matrix UI rollout. Step 1 (#547) introduced a shared tab shell and migrated Individual Tracker Manager. This step adds analytics consumption plumbing and kill matrix presentation logic so UI wiring can stay focused in Step 3.

## 2. overview of the feature
Add a frontend match-analytics service and a kill-matrix presenter layer that transforms API analytics into view-ready rows. Also wire service installation through app service containers used by Discord Series Stats and Individual Tracker Viewer.

## 3. what this PR does
- Adds frontend analytics service contract + real implementation + fake:
  - `pages/src/services/stats/match-analytics-types.ts`
  - `pages/src/services/stats/match-analytics.ts`
  - `pages/src/services/stats/fakes/match-analytics.fake.ts`
  - `pages/src/services/stats/tests/match-analytics.test.ts`
- Extends stats installers:
  - `pages/src/services/stats/install.ts`
- Wires analytics service into app-level service containers:
  - `pages/src/apps/discord-series-stats/services.ts`
  - `pages/src/apps/individual-tracker-viewer/services.ts`
  - `pages/src/apps/discord-series-stats/fakes/create.fake.ts`
- Adds kill matrix presenter and types (no UI rendering yet):
  - `pages/src/components/stats/kill-matrix/types.ts`
  - `pages/src/components/stats/kill-matrix/kill-matrix-presenter.ts`
  - `pages/src/components/stats/kill-matrix/tests/kill-matrix-presenter.test.ts`

Validation:
- `npx vitest run pages/src/services/stats/tests/match-analytics.test.ts pages/src/components/stats/kill-matrix/tests/kill-matrix-presenter.test.ts pages/src/apps/discord-series-stats/tests/create.test.tsx pages/src/apps/discord-series-stats/tests/discord-series-stats-presenter.test.ts pages/src/components/discord-series-stats/tests/create.test.tsx`
- `npm run typecheck:pages`

## 4. PR train
| Step | Branch | PR | Scope | Status |
|---|---|---|---|---|
| 1 | `feature/stage-2b-kill-matrix-ui` | #547 | Shared tab shell + Individual Tracker Manager migration + tests | Open |
| 2 | `feature/stage-2b-analytics-service` | this PR | Match analytics service plumbing + kill matrix presenter + tests | Open |
| 3 | `feature/stage-2b-kill-matrix-table` | pending | Kill matrix table UI and wiring into Match Stats, Series Stats, Individual Tracker Viewer + tests | Planned |

### CODE REVIEW

File: pages/src/services/stats/fakes/match-analytics.fake.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export class FakeMatchAnalyticsService implements MatchAnalyticsService {
+  private readonly analytics: MatchAnalytics;
+
+  constructor(options: Partial<FakeMatchAnalyticsServiceOptions> = {}) {
+    this.analytics = options.analytics ?? aFakeMatchAnalyticsWith();
+  }
+
+  async getMatchAnalytics(): Promise<MatchAnalytics> {
+    return Promise.resolve(this.analytics);
+  }
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply pages/src/services/stats/fakes/match-analytics.fake.ts]: Fixed in 614cf8a. Updated FakeMatchAnalyticsService.getMatchAnalytics to accept matchId/modules and explicitly void unused parameters to match MatchAnalyticsService.

File: pages/src/apps/discord-series-stats/fakes/create.fake.ts [hunk 2]
```diff
@@ -108,5 +126,10 @@ export function aFakeDiscordSeriesStatsAppServicesWith(response: DiscordSeriesSt
         return Promise.resolve({ status: 200, retryAfterSeconds: null });
       },
     },
+    matchAnalyticsService: {
+      getMatchAnalytics: async (): Promise<MatchAnalytics> => {
+        return Promise.resolve(aFakeMatchAnalyticsWith());
+      },
+    },
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply pages/src/apps/discord-series-stats/fakes/create.fake.ts]: Fixed in 614cf8a. Updated the discord-series-stats fake matchAnalyticsService.getMatchAnalytics signature to accept matchId/modules and void unused parameters for contract consistency.

File: pages/src/components/stats/kill-matrix/kill-matrix-presenter.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+function topWeaponId(weapons: readonly { readonly weaponId: number; readonly count: number }[]): number | null {
+  if (weapons.length === 0) {
+    return null;
+  }
+
+  const sorted = [...weapons].sort((a, b) => b.count - a.count);
+  return sorted[0]?.weaponId ?? null;
+}
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply pages/src/components/stats/kill-matrix/kill-matrix-presenter.ts]: Fixed in 1c243fa. Replaced O(n log n) sort with O(n) single-pass max scan to find the weapon with highest count.

File: pages/src/services/stats/match-analytics.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  async getMatchAnalytics(
+    matchId: string,
+    modules: readonly AnalyticsModule[] = DEFAULT_MODULES,
+  ): Promise<MatchAnalytics> {
+    const query = new URLSearchParams({ modules: buildModulesQuery(modules) });
+    const response = await fetch(`${this.apiHost}/api/stats/match-analytics/${matchId}?${query.toString()}`, {
+      credentials: "include",
+    });
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply pages/src/services/stats/match-analytics.ts:34]: Fixed in 1c243fa. Added check to normalize empty modules array to DEFAULT_MODULES before building query to avoid 400 errors from backend.

File: pages/src/components/stats/kill-matrix/kill-matrix-presenter.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+  for (const weapon of weapons) {
+    if (weapon.count > maxCount) {
+      maxCount = weapon.count;
+      maxWeaponId = weapon.weaponId;
+    }
+  }
+
+  return maxWeaponId;
+}
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply pages/src/components/stats/kill-matrix/kill-matrix-presenter.ts]: Fixed in e95928a. Initialize max scan to first entry to guarantee topWeaponId returns a weapon even when all counts are zero. Added test to verify this edge case.

File: pages/src/services/stats/match-analytics.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    this.apiHost = apiHost;
+  }
+
+  async getMatchAnalytics(
+    matchId: string,
+    modules: readonly AnalyticsModule[] = DEFAULT_MODULES,
+  ): Promise<MatchAnalytics> {
+    const normalizedModules = modules.length === 0 ? DEFAULT_MODULES : modules;
+    const query = new URLSearchParams({ modules: buildModulesQuery(normalizedModules) });
+    const response = await fetch(`${this.apiHost}/api/stats/match-analytics/${matchId}?${query.toString()}`, {
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply pages/src/services/stats/match-analytics.ts]: Fixed in 13f5502. Encoded matchId via encodeURIComponent before URL path interpolation and added a regression test for reserved characters in matchId.
